admin/includes/functions/html_output.php

```diff
@@ -503,34 +503,49 @@ function zen_draw_date_selector($fieldname_prefix, $default_date='') {
     return $date_selector;
 }
 
-function addSearchKeywordForm($filename, $action) {
-    $keywords_products = (isset($_POST['keywords']) && zen_not_null($_POST['keywords'])) ? zen_db_input(zen_db_prepare_input($_POST['keywords'])) : '';
-    $fullAction = '';
-    if (!empty($action)) {
-        $fullAction = 'action=' . $action;
-    }
-    $form = '
+    /**
+     * intended to add a universal search for use with any pulldown that extends the
+     * abstract pulldown class.  returns a html string.
+     * @param string $filename
+     * @param string $action
+     * @param bool $includeForm
+     * @param array $extrafieldsArray
+     * @return string
+     */
+
+    function addSearchKeywordForm(string $filename = '', string $action = '', bool $includeForm = true, array $extrafieldsArray = [])
```

Comment:
Why are there now defaults for the `$filename` and `$action` parameters?  I can 'see' the use for a default/null action, but not the `$filename`, since it's used in both the (now optional) form but also in the 'keyword products' link.

Reply:
i agree that `$filename` should be made mandatory.